### PR TITLE
Refactored lobby page to stop weird scrolling and make the layout more responsive

### PR DESCRIPTION
**Pull request by tillywoodfield**

### CODE REVIEW

File: src/components/Lobby/Header/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import styles from './Header.module.css';
+import Logo from '../../Logo/Logo';
+import Logout from '../../Logout/Logout';
+import LeaderboardButton from '../../Leaderboard/LeaderboardButton/LeaderboardButton';
+import UploadButton from '../../Upload/UploadButton/UploadButton';
+
+export class Header extends React.Component {
```
Comment (Ro4052) [inline src/components/Lobby/Header/Header.jsx]: Might as well make this a functional component.

File: src/components/Upload/Upload.jsx [hunk 2]
```diff
@@ -16,6 +17,10 @@ export class Upload extends React.Component {
         this.onChange = this.onChange.bind(this);
     }
 
+    goToLobby() {
+        history.push('/lobby');
```
Comment (Ro4052) [inline src/components/Upload/Upload.jsx]: I thought we'd gone with the approach of using the `history` object passed down through the tree?

**Review by Ro4052 — DISMISSED** [review]

File: src/components/Upload/Upload.jsx (hunk 2 above)
Comment (tillywoodfield) [reply src/components/Upload/Upload.jsx]: I just copied this function from the leaderboard page, but happy to change it?

Comment (Ro4052) [reply src/components/Upload/Upload.jsx]: That's up to you. Another point in a similar area though, is the `UploadButton` class ever being used?

File: src/components/Upload/UploadButton/UploadButton.jsx [hunk 3]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
-import history from '../../../services/history';
+import { withRouter } from "react-router";
 
 export class UploadButton extends React.Component {
     goToUpload() {
-        history.push('/upload');
+
```
Comment (Ro4052) [inline src/components/Upload/UploadButton/UploadButton.jsx]: Added an extra line, also, is this file actually used?

**Review by Ro4052 — DISMISSED** [review]

**Comment by Ro4052** [comment]
I keep accidently clicking approve instead of comment. One of the tests is failing anyway...

File: src/components/Upload/UploadButton/UploadButton.jsx (hunk 3 above)
Comment (tillywoodfield) [reply src/components/Upload/UploadButton/UploadButton.jsx]: The history file is still used in the redux actions

**Review by Ro4052 — APPROVED** [review]
This time I _meant_ to approve.

**Pull request merged**